src/lib/core/services.js

```diff
@@ -47,9 +49,54 @@ export function getServiceInformation(name) {
   throw new Error('Unknown Service'); // TODO: Check Error
 }
 
-// TODO: Implement
+export async function getServiceKeysForUser(user) {
+  const list = await database.services.doc(user.id).get();
+
+  if (!list.exists) {
+    return [];
+  }
+
+  return Object.keys(list.data());
+}
+
+export async function getServiceListForUser(user) {
+  log(`Retrieving service list for ${user.username}`);
+
+  const keys = await getServiceKeysForUser(user);
+  return keys.map(key => SERVICE_LIST.get(key).name);
+}
+
 export async function getServiceForUser(user, serviceName) {
-  return `${user.name}#${serviceName}`;
+  log(`Retrieving ${serviceName} for ${user.username}`);
+
+  const serviceList = await database.services.doc(user.id).get();
+
+  if (!serviceList.exists) {
+    return null;
+  }
+
+  const service = serviceList.data();
+
+  return service[serviceName];
+}
+
+export async function setServiceInformationForUser(user, name, fields) {
+  log(`Setting ${name} for ${user.username}`);
+
+  return database.services.doc(user.id).set(
+    {
+      [name]: fields,
+    },
+    { merge: true }
+  );
+}
+
+export async function removeServiceForUser(user, serviceName) {
```

Comment:
Probably don't need all the `service` text in the functions here - it's already namespaces as `aquarius.services.X`